include/libcyphal/application/registry/register_impl.hpp

```diff
@@ -0,0 +1,350 @@
+        : Base{memory, name, options}
+        , getter_(std::move(getter))
+    {
+    }
+
+    ~Register()                   = default;
+    Register(Register&&) noexcept = default;
+
+    Register(const Register&)                = delete;
+    Register& operator=(const Register&)     = delete;
+    Register& operator=(Register&&) noexcept = delete;
+
+    // MARK: IRegister
+
+    ValueAndFlags get() const override
+    {
+        return getImpl(getter_(), false);
+    }
+
+    cetl::optional<SetError> set(const Value&) override
+    {
+        return SetError::Mutability;
+    }
+
+private:
+    // MARK: Data members:
+
+    Getter getter_;
+
+};  // Register<IsMutable=false>
+//
+/// Defines a read-write register implementation.
+///
+/// The actual value is provided by the getter function,
+/// and the setter function is used to update the value.
+///
+template <typename Getter, typename Setter>
+class Register<Getter, Setter, true> final : public RegisterBase
+{
+    using Base = RegisterBase;
+
+public:
+    /// Constructs a new read-write detached register, which is not yet linked to any registry (aka detached).
+    ///
+    /// A detached register must be appended to a registry before its value could be exposed by the registry.
+    ///
+    /// @param memory The memory resource to use for variable size-d register values.
+    /// @param name The name of the register.
+    /// @param options Extra options for the register, like "persistent" option (`true` by default).
+    /// @param getter The getter function to provide the register value (from a storage).
+    /// @param setter The setter function to update the register value (at the storage).
+    ///
+    Register(cetl::pmr::memory_resource& memory,
+             const Name                  name,
+             const Options&              options,
+             Getter&&                    getter,
+             Setter&&                    setter)
+        : Base{memory, name, options}
+        , getter_{std::move(getter)}
+        , setter_{std::move(setter)}
+    {
+    }
+
+    ~Register()                   = default;
+    Register(Register&&) noexcept = default;
+
+    Register(const Register&)                = delete;
+    Register& operator=(const Register&)     = delete;
+    Register& operator=(Register&&) noexcept = delete;
+
+    // MARK: IRegister
+
+    ValueAndFlags get() const override
+    {
+        return getImpl(getter_(), true);
+    }
+
+    cetl::optional<SetError> set(const Value& new_value) override
+    {
+        return setImpl(new_value, setter_);
+    }
+
+private:
+    // MARK: Data members:
+
+    Getter getter_;
+    Setter setter_;
+
+};  // Register<IsMutable=true>
+
+// MARK: -
+
+/// Defines a parameter-based register implementation template.
+///
+/// Instead of "external" getter/setter functions, the register uses member value storage.
+///
+template <typename ValueType, bool IsMutable = true>
+class ParamRegister final : public RegisterBase
```

Comment:
We seem to be missing callable registers that don't store the value but instead construct it ad-hoc; in the original codebase it was done via `route`: https://github.com/Zubax/telega/blob/80f5203d8536b7be7e6fc991a8462b73ea5930cb/telega/src/registry/registry_impl.hpp#L151-L197

This is an essential feature

Reply:
See below comment